libcontainer/apparmor/apparmor.go

```diff
@@ -6,17 +6,28 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"os/exec"
 
 	"github.com/opencontainers/runc/libcontainer/utils"
 )
 
 // IsEnabled returns true if apparmor is enabled for the host.
 func IsEnabled() bool {
 	if _, err := os.Stat("/sys/kernel/security/apparmor"); err == nil && os.Getenv("container") == "" {
-		if _, err = os.Stat("/sbin/apparmor_parser"); err == nil {
+		isEnabledByParameters := func() bool {
 			buf, err := ioutil.ReadFile("/sys/module/apparmor/parameters/enabled")
 			return err == nil && len(buf) > 1 && buf[0] == 'Y'
 		}
+
+		// /sbin/ is not always in $PATH, so we check it explicitly
+		if _, err = os.Stat("/sbin/apparmor_parser"); err == nil {
+			return isEnabledByParameters()
+		}
+
+		// Fallback to checking $PATH
+		if _, err = exec.LookPath("apparmor_parser"); err == nil {
+			return isEnabledByParameters()
+		}
```

Comment:
I'm not sure that these checks are actually necessary -- the logic is "well, a box without `apparmor_parser` isn't a working AppArmor distribution" but I don't think that makes sense to enforce that in runc. All we care about is whether AppArmor is enabled in the kernel -- and if the user isn't actually using the profiles then the `config.json` won't contain an AppArmor profile setting anyway.

This check comes from Docker, which *does* use `apparmor_parser` (so this check makes a little bit more sense there) but we don't in runc so I think it'd just be simpler to drop these checks and just check `/sys/kernel` and `/sys/module`.

And as an aside, while we aren't executing any binaries here, `$PATH` lookups in runc make me nervous.

Reply:
Yes I agree, I think we could omit the check for `apparmor_parser` and let higher level runtimes do this work.